Rename concurrent_prepare to two_write_queues

maysamyabandeh opened the pull request:
Summary: concurrent_prepare in rocksdb is renamed to two_write_queues.
This patch updates the rocksdb submodule and make similar change in
myrocks.
@update-submodule: rocksdb

lth commented:
`rocksdb_sys_vars.all_vars` is failing because you need to rename `mysql-test/suite/rocksdb_sys_vars/t/rocksdb_concurrent_prepare_basic.test`.

laurynas-biveinis commented on mysql-test/r/mysqld--help-notwin-profiling.result:
```diff
@@ -1162,7 +1162,7 @@ The following options may be given as the first argument:
  Possible values are ON, OFF, FORCE (don't start if the
  plugin fails to load).
  --rocksdb-concurrent-prepare 
```
This is suspicious - the variable help text is changed but not the variable name itself. Likewise re. --skip-rocksdb-concurrent-prepare two lines down. Does the testcase pass?

@georgelorchpercona, does the var rename make it to our GA freeze?

maysamyabandeh replied:
I am going to keep the myrocks var name. We only need to update the rocksdb var.

george-lorch replied:
@laurynas-biveinis, no, this will not likely make the PS 5.7 MyRocks GA. This PR hasn't even landed yet. If it did tomorrow, it would still be be within the November merge group which is not planned to be included in the GA. We are currently planning on releasing GA merged up through September or October, with October being a bit of a stretch. So we are effectively already frozen. The change, as originally authored, would eventually be reviewed and discarded or merged in a fixed up manner that would not violate our compatibility policy. As stated many times over the last year, once we ship GA, we will no longer merge into our 5.7 tree any changes that introduce renaming of variables or alterations of default behaviors of variables in a mid-series release. Users demand that mid-series upgrades just work without needing to alter configurations or their applications. Any such changes will have to wait until we port to 8.0 or we would need very explicit direction from Vadim and/or PZ to override that policy.

george-lorch replied:
These kinds of things are also problematic in rocksdb, not just myrocks, if an option name is changed with no aliasing, rocksdb will fail to start until the user edits the rocksdb OPTIONS file and corrects it or uses the recently added "fb-mysql-5.6.35/13d2e6f2da9808ad7b4cfe86394aa56734572e7d : Enable unknown options being passed to rocksDB". This is terrible practice to alter option names and default values and expect users to know and understand that they need to perform manual steps on upgrades.

Here is a case already that illustrates what an end user potentially could hit if options are changed and released into the wild with no backward compatibility or automatic upgrade. I do concede though that his particular instance looks like the user is attempting to shuffle variants without fully cleaning their datadir in between installs, but you get the idea of what we will run into. https://groups.google.com/forum/#!topic/percona-discussion/JEF4njF1094

maysamyabandeh replied:
at rocksdb the old var name is still accepted but simply ignored. the "enable unknown option" is only for forward compatibility: when you downgrade to an earlier version where the old binary does not understand the options added by the new binary.

lth replied:
We added `ignore_unknown_options` for the purpose of downgrading, with the understanding that it's okay if miss setting some of these options on an older version. Upgrades should work just fine.

Since this is going to cause problems anyway with the OPTIONS file, I think we should make a clean break and rename the MyRocks variable as well, so it's clear to the user that a variable rename has occurred.

laurynas-biveinis commented on mysql-test/r/mysqld--help-notwin.result:
```diff
@@ -1160,7 +1160,7 @@ The following options may be given as the first argument:
  Possible values are ON, OFF, FORCE (don't start if the
  plugin fails to load).
  --rocksdb-concurrent-prepare 
```
Likewise

laurynas-biveinis commented on mysql-test/suite/rocksdb/r/rocksdb.result:
```diff
@@ -879,7 +879,7 @@ rocksdb_compaction_sequential_deletes	0
 rocksdb_compaction_sequential_deletes_file_size	0
 rocksdb_compaction_sequential_deletes_window	0
-rocksdb_concurrent_prepare	ON
+rocksdb_two_write_queues	ON
```
The list looks sorted. Does the testcase pass?

maysamyabandeh replied:
Thanks. I just noticed the test failure. I rushed into submitting the PR. I will submit a fix soon.

Commit:
Rename concurrent_prepare to two_write_queues

Summary: concurrent_prepare in rocksdb is renamed to two_write_queues.
This patch updates the rocksdb submodule and make similar change in
myrocks.
@update-submodule: rocksdb
